ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh

```diff
@@ -60,6 +58,48 @@ fi
 
 echo "Detected PG version: $PGVERSION"
 
+function shim_extensions {
+    PKGLIB=$1
+    # This is a workaround for older versions of wrappers which don't have the expected
+    #  naming scheme, containing the version in their library's file name
+    #  e.g. wrappers-0.1.16.so, rather than wrappers.so
+    # pg_upgrade errors out when it doesn't find an equivalent file in the new PG version's
+    #  library directory, so we're making sure the new version has the expected (old version's)
+    #  file name.
+    # After the upgrade completes, the new version's library file is used.
+    # i.e. 
+    #  - old version: wrappers-0.1.16.so
+    #  - new version: wrappers-0.1.18.so
+    #  - workaround to make pg_upgrade happy: copy wrappers-0.1.18.so to wrappers-0.1.16.so
```

Comment:
this feels so wrong, and is going to be hard to debug post-hoc. I take it that symlinks don't work?

If they don't, maybe we can touch a hidden file indicating what we've done as part of the upgrade?

Reply:
Symlinks should work, as they're regularly used by lib files. Switching to use symlinks